### PR TITLE
Fix warnings in DispatchQueue.sync() implementation when using a comp…

### PR DESCRIPTION
**Pull request by ktopley-apple**
…iler with SE-0176 support.

(Radar 32888663)

### CODE REVIEW

File: test/stdlib/Dispatch.swift [hunk 1]
```diff
@@ -40,6 +40,13 @@ DispatchAPI.test("DispatchGroup creation") {
   expectNotNil(group)
 }
 
+DispatchAPI.test("Dispatch sync return value") {
+	let value = 24;
```
Comment (moiseev) [inline test/stdlib/Dispatch.swift]: Minor, but the rest of this file seems to be using 2 spaces for indentation.

**Comment by ktopley-apple** [comment]
The corresponding change for the overlay is at https://github.com/apple/swift-corelibs-libdispatch/pull/268

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by ktopley-apple** [comment]
@swift-ci Please smoke test

**Comment by ktopley-apple** [comment]
Fixed test file indent per Max's review comment.

**Comment by rjmccall** [comment]
Thank you, that seems fine to me.

**Comment by moiseev** [comment]
@swift-ci Please test and merge

**Review by das — APPROVED** [review]
👍🏻
I merged the matching PR for swift-corelibs-libdispatch

**Pull request merged**